pkg/guard-utils/santize.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package guardutils
+
+// contain at most 63 characters
+// contain only lowercase alphanumeric characters or '-'
+// start with an alphanumeric character
+// end with an alphanumeric character
+func Sanitize(in string) string {
```

Comment:
I'd separate the different sub-sections into helper functions (easy) and Sanatize just calls the others. Allows for easy testing and easier modification if Sanatize needs to be enhanced...

Reply:
let's let the natural course of things define what other sanitization is needed and build helpers as needed and where needed.